File: Php/app/Urls/HomegameJoinUrlModel.php
<?php
namespace app\Urls{

	use app\RouteFormats;
	use app\Urls\BaseClasses\HomegameUrlModel;
	use entities\Homegame;

	class HomegameJoinUrlModel extends HomegameUrlModel{

		public function __construct(Homegame $homegame){
			parent::__construct(RouteFormats::homegameJoin, $homegame);
		}

	}

}

File: Web/Models/UrlModels/HomegameJoinUrlModel.cs
using Core.Classes;
using Web.Routing;

namespace Web.Models.UrlModels{

	public class HomegameJoinUrlModel : HomegameUrlModel{

		public HomegameJoinUrlModel(Homegame homegame)
            : base(RouteFormats.HomegameJoin, homegame)
        {
		}

	}

}